RxSwift/Deprecated.swift

```diff
@@ -35,6 +35,75 @@ extension Observable {
     }
 }
 
+extension ObservableType {
+    /**
+
+    ** @available(*, deprecated, message: "Please use enumerated().map()", renamed: "enumerated().map()") **
+
+     Projects each element of an observable sequence into a new form by incorporating the element's index.
+
+     - seealso: [map operator on reactivex.io](http://reactivex.io/documentation/operators/map.html)
+
+     - parameter selector: A transform function to apply to each source element; the second parameter of the function represents the index of the source element.
+     - returns: An observable sequence whose elements are the result of invoking the transform function on each element of source.
+     */
+    public func mapWithIndex<R>(_ selector: @escaping (E, Int) throws -> R)
+        -> Observable<R> {
+        return enumerated().map { try selector($0.element, $0.index) }
```

Comment:
Do you think this should be flipped? index first, element second, like Swift's `enumerated()` method on Sequences ?